posts/basic-probabliity-in-lean.md

```diff
@@ -0,0 +1,245 @@
+- Expectation of `X` under `P`: `P[X]`.
+- Variance: `variance X P`.
+- Moment of order `p`: `moment X p P`.
+- Central moment of order `p`: `centralMoment X p P`.
+- Moment generating function of `X` under `P` at `t : ℝ`: `mgf X P t`.
+- Cumulant generating function: `cgf X P t`.
+
+### Known probability distributions
+
+The Probability/Distributions folder of Mathlib contains several common probability laws: Exponential, Gamma, Gaussian (only in `ℝ`), Geometric, Pareto, Poisson, Uniform.
+Other measures that can be used to build probability distributions are defined elsewhere, in the MeasureTheory folder: counting measure, Dirac, Lebesgue measure.
+
+For example, the Gaussian distribution on the real line with mean `μ` and variance `v` is a `Measure ℝ`:
+```lean
+def gaussianReal (μ : ℝ) (v : ℝ≥0) : Measure ℝ := -- omitted
+```
+The definition is followed by an instance `IsProbabilityMeasure (gaussianReal μ v)` that states that this is a probability measure. It is defined as the Dirac distribution for `v = 0`.
+The file where the Gaussian is defined also contains properties of its p.d.f.
+
+As another example, to take the uniform distribution on a finite set `s : Set Ω`, use `uniformOn s : Measure Ω`.
+
+### Identically distributed
+
+`IdentDistrib X Y P Q` (or `IdentDistrib X Y` in `MeasureSpace`) states that `X` and `Y` are identically distributed.
+
+### Conditioning
+
+#### Conditional probability
+
+The probability of a set `s` conditioned on another set `t` for the measure `P` is `P[s|t]`, which equals `P (s ∩ t) / P t`.
+Conditioning on `t` defines a new measure named `cond P t`, denoted by `P[|t]`. With that notation, `P[s|t] = P[|t] s`.
+
+For `X : Ω → E` and `x : E`, we write `P[|X ← x]` for the probability measure conditioned on `X = x`. It is notation for `P[|X ⁻¹' {x}]`.
+This is meaningful only if `X = x` has non-zero probability, so that definition is mostly useful for discrete probability.
+
+#### Conditional expectation
+
+The conditional expectation of a random variable `Y : Ω → F` given a sigma-algebra `m` of `Ω` is `condexp m P Y`, with notation `P[Y | m]`. This is a random variable `Ω → F`.
+The conditional expectation of `Y` given `X : Ω → E` can be obtained with `P[Y | mE.comap X]`, in which `mE : MeasurableSpace E` is the sigma-algebra on `E`.
+
+We have special notation for the conditional expectation of the real indicator of a set `s : Set Ω`: `P⟦s | m⟧ : Ω → ℝ`.
+
+#### Conditional probability distribution
+
+The regular conditional probability distribution of `Y : Ω → F` given `X : Ω → E`, for `F` standard Borel, is denoted by `condDistrib Y X P` (for a measure `P` on `Ω`).
+This is a Markov kernel from `E` to `F` (see further down) such that for all measurable sets `s` of `F`, for `P`-almost every `ω : Ω `, `condDistrib Y X P (X ω) s` is equal to `P⟦Y ⁻¹' s|mE.comap X⟧` (up to a mismatch in types: `ℝ≥0∞` for `condDistrib` versus `ℝ` for the conditional expectation of the indicator).
+
```

Comment:
Add a comment somewhere that this only makes sense in standard Borel spaces? (At other places of the blog post also, maybe?)

Reply:
There is "for `F` standard Borel" in the first sentence of that paragraph, so I alluded to it a bit already. Do you want me to be more explicit about it?

Reply:
I added a remark about standard Borel in the conditional independence section.

Reply:
You read my mind: this remark came to me while I was reading the paragraph on conditional independence (where standard Borel was not mentioned), but then I thought I should mention it at the first place it was relevant, and I didn't notice it was indeed mentioned there. Sorry!